### PR TITLE
vpv: init at 0.8.1

### PR DESCRIPTION
**Pull request by kidanger**
###### Description of changes

new package: https://github.com/kidanger/vpv (scientific image visualization tool)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by minijackson — APPROVED** [review]
Looks good, and seems to work great. Thanks!

---

Result of `nixpkgs-review pr 229180` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vpv</li>
  </ul>
</details>

File: pkgs/applications/graphics/vpv/default.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+, libpng
+, libtiff
+, libjpeg
+, SDL2
+, gdal
+, octave
+, rustPlatform
+, cargo
+}:
+stdenv.mkDerivation rec {
```
[the remark this answers is not in the document]
Comment (kidanger) [reply pkgs/applications/graphics/vpv/default.nix]: Thanks for the suggestion, I've applied the change.

File: pkgs/applications/graphics/vpv/default.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    "-DUSE_GDAL=ON"
+    "-DUSE_OCTAVE=ON"
+    "-DVPV_VERSION=v${finalAttrs.version}"
+    "-DBUILD_TESTING=ON"
+  ];
+
+  doCheck = true;
+  checkTarget = "test";
+
+  meta = with lib; {
```
Comment (drupol) [inline pkgs/applications/graphics/vpv/default.nix:59]:
```suggestion
  meta = {
```

File: pkgs/applications/graphics/vpv/default.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    homepage = "https://github.com/kidanger/vpv";
+    description = "Image viewer for image processing experts";
+    maintainers = with maintainers; [ kidanger ];
```
Comment (drupol) [inline pkgs/applications/graphics/vpv/default.nix:62]:
```suggestion
    maintainers = [ lib.maintainers.kidanger ];
```

File: pkgs/applications/graphics/vpv/default.nix [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    license = licenses.gpl3;
```
Comment (drupol) [inline pkgs/applications/graphics/vpv/default.nix:63]:
```suggestion
    license = lib.licenses.gpl3;
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Review by drupol — APPROVED** [review]
LGTM !

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -8161,6 +8161,11 @@
     githubId = 16481032;
     name = "Kiba Fox";
   };
+  kidanger = {
+    github = "kidanger";
+    githubId = 297479;
+    name = "Jérémy Anger";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8167]: It wouldn't hurt to add another mean of contact

File: pkgs/applications/graphics/vpv/default.nix [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  buildInputs = [
+    libpng
+    libtiff
+    libjpeg
+    SDL2
+    gdal
+    octave
+    rustPlatform.cargoSetupHook
+    cargo
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/vpv/default.nix:46]: Those two go to nativeBuildInputs

Comment (kidanger) [reply pkgs/applications/graphics/vpv/default.nix]: Fixed, thanks.

File: maintainers/maintainer-list.nix (hunk 5 above)
Comment (kidanger) [reply maintainers/maintainer-list.nix:8168]: I've added the email information.

File: pkgs/applications/graphics/vpv/default.nix [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  ];
+
+  cmakeFlags = [
+    "-DUSE_GDAL=ON"
+    "-DUSE_OCTAVE=ON"
+    "-DVPV_VERSION=v${finalAttrs.version}"
+    "-DBUILD_TESTING=ON"
+  ];
+
+  doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/vpv/default.nix:56]: fails to build

```
---- tests::test_fuzzy_path_finder_multiple_results_spaces stdout ----
thread 'tests::test_fuzzy_path_finder_multiple_results_spaces' panicked at 'assertion failed: `(left == right)`
  left: `["fo ob arimage.png", "foobarm age.p ng", "f oobarimae.jpg"]`,
 right: `["fo ob arimage.png", "foobarm age.p ng"]`', src/lib.rs:300:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Comment (kidanger) [reply pkgs/applications/graphics/vpv/default.nix]: Indeed the tests sometimes fail, the issue is known but I forgot about it. I've removed the `doCheck`, and I will put it back once this is fixed upstream.

**Pull request merged**